source/application/AMPIC887DriverServer/AMPIC887ConsoleApplication.cpp

```diff
@@ -0,0 +1,117 @@
+#include "AMPIC887ConsoleApplication.h"
+
+#include <QTimer>
+
+#include "AMConsoleInputHandler.h"
+#include "AMPIC887ConsoleCommandParser.h"
+#include "AMGCS2Support.h"
+
+AMPIC887ConsoleApplication::AMPIC887ConsoleApplication(int argc, char *argv[]) :
+	QCoreApplication(argc, argv)
+{
+
+	// Set up the console and command set.
+	commandParser_ = new AMPIC887ConsoleCommandParser();
+	consoleInputHandler_ = new AMConsoleInputHandler(commandParser_, this);
+
+
+	if(startup()) {
+
+		// Print a welcome message
+		consoleInputHandler_->setCommandPrompt(
+					QString("%1 > ")
+					.arg(controllerCollection_.activeController()->name()));
+
+		consoleInputHandler_->writeLineToStandardOutput("Welcome to the PI C887 Server Application.");
+		consoleInputHandler_->writeLineToStandardOutput(controllerCollection_.connectionStatuses());
+		consoleInputHandler_->writeLineToStandardOutput("Type 'help' for a list of commands.");
+		consoleInputHandler_->displayCommandPrompt();
+		connect(commandParser_, SIGNAL(help()), this, SLOT(onHelpCommandIssued()));
+		connect(commandParser_, SIGNAL(quit()), this, SLOT(onQuitCommandIssued()));
+		connect(commandParser_, SIGNAL(status()), this, SLOT(onStatusCommandIssued()));
+		connect(commandParser_, SIGNAL(changeActiveController(QString)), this, SLOT(onActiveControllerChangeIssued(QString)));
+		connect(commandParser_, SIGNAL(GCS2CommandIssued(AMGCS2Command*)), this, SLOT(onGCS2CommandIssued(AMGCS2Command*)));
+
+	} else {
+
+		// Startup failed. Print an error message, then startup a timer to close
+		// the application after a few seconds.
+		consoleInputHandler_->writeLineToStandardError("Could not startup the hexapod driver application.");
+
+		QTimer* exitTimer = new QTimer(this);
```

Comment:
In this case it needs to output to the console. Should require significantly less time than this. I meant to come back later an experiment with lower values, but forgot.